### ISSUE TITLE
Crop without scaling up?

### ISSUE THREAD

**Issue by eveliotc** [thumbor/thumbor#267]
I'm trying to get this behaviour out of thumbor, say I have [this image](https://pbs.twimg.com/profile_images/378800000822867536/3f5a00acf72df93528b6bb7cd0a4fd0c.jpeg) which is 537x529 and 32K, 

![https://pbs.twimg.com/profile_images/378800000822867536/3f5a00acf72df93528b6bb7cd0a4fd0c.jpeg](https://pbs.twimg.com/profile_images/378800000822867536/3f5a00acf72df93528b6bb7cd0a4fd0c.jpeg)

And my view has a 16:9 ratio, so assuming we are at 1x we use that ratio which based on width would be 537x302, so we request

`http://thumbor/unsafe/537x302/pbs.twimg.com/profile_images/378800000822867536/3f5a00acf72df93528b6bb7cd0a4fd0c.jpeg`

So we get a nice cropped image of 537x302 which is 18K

![3f5a00acf72df93528b6bb7cd0a4fd0c 1](https://f.cloud.github.com/assets/400641/2339544/05520c9c-a4b7-11e3-912c-9764b9ec8d1d.jpeg)

So far so good, problem is when we go to higher densities and views start getting bigger and bigger so in example those 537x302 in a higher density which is lets say is 4x are 2148x1208, so we request such size as

`http://thumbor/unsafe/2148x1208/pbs.twimg.com/profile_images/378800000822867536/3f5a00acf72df93528b6bb7cd0a4fd0c.jpeg`

And we get our 2148x1208 image which is 103K, around 3x as original image, which makes sense as we are requesting a bunch more pixels

![3f5a00acf72df93528b6bb7cd0a4fd0c 4](https://f.cloud.github.com/assets/400641/2339649/6800cd78-a4b8-11e3-95e4-8d282d78c05f.jpeg)

But I'd like thumbor to return me the maximum size available cropped out **without scaling up**. In this example would be the 537x302 18K image.

As I don't know the size of the image beforehand, I can't actually compute the dimensions based on base (1x) density, and even if I had them I don't want to miss the chance of getting a better image quality if available, the behaviour I'm trying to get out of thumbor is one in which the image does not scale up so KB size is small as possible yet we get our image cropped out, in example we request

`http://thumbor/unsafe/somethingThatDisablesScaleUp/2148x1208/pbs.twimg.com/profile_images/378800000822867536/3f5a00acf72df93528b6bb7cd0a4fd0c.jpeg`

And we get an image of 537x302 which is 18K, as the image is not larger than 537px with `somethingThatDisablesScaleUp` even if given size is larger (2148x1208), we don't scale up and we return the maximum size based on ratio without scaling up at all.

I noticed `fit-in` has a similar behaviour but there is no way to crop, Is there any way I could achieve such behaviour?

**Comment by chrisvis** [comment]
Using `no_upscale()` I seem to indeed get the original size when requesting a bigger version. So if the original size is 100x100 and I request a 200x400, I get a 100x100 back. Ideally I would like a non-upscaled version of the original but cropped to the ratio I requested, in this case I requested 200x400, so a ratio of 1:2 and I would like to get an image of size 50x100. Is there a way to achieve this?
